jenkins/gradle/gradle-check.jenkinsfile

```diff
@@ -0,0 +1,123 @@
+    agent {
+        node {
+            // Must use Ubuntu agent with 1 executor or gradle check will show a lot of java-related errors
+            label 'Jenkins-Agent-Ubuntu2004-X64-c54xlarge-Single-Host'
+        }
+    }
+    parameters {
+        string(
+            name: 'GIT_REPO_URL',
+            description: 'OpenSearch core repository url on git, can be either the official upstream url or your fork url.',
+            defaultValue: 'https://github.com/opensearch-project/OpenSearch.git',
+            trim: true
+        )
+        string(
+            name: 'GIT_REFERENCE',
+            description: 'Git branch, tag, commitid for reference to checkout commit of OpenSearch core before running the gradle check.',
+            defaultValue: 'main',
+            trim: true
+        )
+    }
+    triggers {
+        GenericTrigger(
+            genericVariables: [
+                [key: 'pr_from_sha', value: '$.pr_from_sha'],
+                [key: 'pr_from_clone_url', value: '$.pr_from_clone_url'],
+                [key: 'pr_to_clone_url', value: '$.pr_to_clone_url'],
+                [key: 'pr_title', value: '$.pr_title'],
+                [key: 'pr_number', value: '$.pr_number']
+            ],
+            tokenCredentialId: 'jenkins-gradle-check-generic-webhook-token',
+            causeString: 'Triggered by PR on OpenSearch core repository',
+            printContributedVariables: false,
+            printPostContent: false
+        )
+    }
+    environment {
+        JAVA11_HOME="/var/jenkins/tools/hudson.model.JDK/jdk-11"
+        JAVA17_HOME="/var/jenkins/tools/hudson.model.JDK/jdk-17"
+        JAVA8_HOME="/var/jenkins/tools/hudson.model.JDK/jdk-8"
+        JAVA_HOME="/var/jenkins/tools/hudson.model.JDK/jdk-17"
+        USER_BUILD_CAUSE = currentBuild.getBuildCauses('hudson.model.Cause$UserIdCause')
+        TIMER_BUILD_CAUSE = currentBuild.getBuildCauses('hudson.triggers.TimerTrigger$TimerTriggerCause')
+    }
+    stages {
+        stage('Install jdk-8') {
+            tools {
+                jdk 'jdk-8'
+            }
+            steps {
+                echo "Install jdk-8"
+                sh("ls ${JAVA8_HOME}/.. | grep jdk-8")
+            }
+        }
+        stage('Install jdk-11') {
+           tools {
+               jdk 'jdk-11'
+           }
+            steps {
+                echo "Install jdk-11"
+                sh("ls ${JAVA11_HOME}/.. | grep jdk-11")
+            }
+        }
+        stage('Install jdk-17') {
+           tools {
+               jdk 'jdk-17'
+           }
+            steps {
+                echo "Install jdk-17"
+                sh("ls ${JAVA17_HOME}/.. | grep jdk-17")
+            }
+        }
+        stage('Run Gradle Check') {
+            steps {
+                script {
+                    sh("ls /var/jenkins/tools/hudson.model.JDK/ && env | grep JAVA")
+                    if (env.USER_BUILD_CAUSE.equals('[]') && env.TIMER_BUILD_CAUSE.equals('[]')) {
```

Comment:
Can we discuss around this check? What purpose does it serves?

Reply:
Just as the variable suggest it is checking what trigger type/cause is it.